docs/api/en/core/Object3D.html

```diff
@@ -221,7 +221,11 @@ <h3>[method:this applyQuaternion]( [param:Quaternion quaternion] )</h3>
 		<p>Applies the rotation represented by the quaternion to the object.</p>
 
 		<h3>[method:this attach]( [param:Object3D object] )</h3>
-		<p>Adds *object* as a child of this, while maintaining the object's world transform.</p>
+		<p>Adds *object* as a child of this, while maintaining the object's world transform. 
+		<br /><br />	
+		Note: If the old or new parent of the object is scaled non-uniformly, the exact visible transformation will not be preserved.
```

Comment:
Uhm, that is a bit over-stated and not necessarily true...

@Mugen87 Do you think we need to add such disclaimers for all the methods that do not fully-support non-uniform scale? If so, we need to identify those methods, and come to an agreement on how to word this disclaimer precisely.

Reply:
Maybe we could add a note about non-uniform scale to the [FAQ](https://threejs.org/docs/#manual/en/introduction/FAQ)?

Reply:
@looeee That sounds good to me!

In general, I'd like to avoid adding such notes to all relevant methods. Having a single place in the documentation that warns about the implications when using non-uniform scale is sufficient to me.

Reply:
Sidenote: If I could start from scratch, I would define `scale` as a scalar value that must be positive. I know this would be a more restrictive implementation but also made things much easier.

Reply:
So just to give you my perspective: I was working with this method and specifically went back to this part of the docs to see if there was any note about the limitations. Not finding it led me down the wrong path for a while.

I like the idea of there being a note on this somewhere in the docs (e.g. FAQ), and for every relevant method we just link to that note?

Reply:
I don't vote to do that. If anything I would link to the FAQ entry from `Object3D.scale`.

Reply:
Let me give you an example why I don't like this approach: If we follow your logic we would have to add a note to each light page that it does not affect objects with `MeshBasicMaterial`. However, it is more appropriate to highlight at the basic material page _once_ that it is an unlit material and thus not affected by light.

Reply:
Sure, I see your logic. Although the non-uniform scaling issue is quite unintuitive compared to unlit materials so it might help the average user to expose the explanation more often.

Reply:
>The problem is not the method but how you define Object3D.scale.

I think the problem _is_ the method. The method assumes the matrix is decomposable; it is not, necessarily.